### PR TITLE
Email doi update

### PR DESCRIPTION
**Pull request by daisieh**
Fills in the title and DOI URL in emails.

### CODE REVIEW

**Comment by daisieh** [comment]
The emails will now look like this:

```
Dear D H,

Thank you for your submission to the Dryad repository. Your data package has been approved, and will be available in the repository once the associated article has been published. 

Title: Data from: aTRAM - automated target restricted assembly method: a fast method for assembling loci across divergent taxa from next-generation sequencing data
DOI: doi:10.5072/FK2dryad.8s4d6
Journal manuscript number: none available

You will receive another email when your data package is released. If you did not request a data embargo, your file(s) will be available for download after the article is published.


CITING YOUR DRYAD DATA

Ensure that readers can find your data!

We recommend that the article include a link to the Dryad data as follows:

Data available from the Dryad Digital Repository: http://dx.doi.org/10.5072/FK2dryad.8s4d6

Your journal may follow the CrossRef recommendation that the data package be cited in the article Reference list. If so, the citation should include the Dryad package DOI, e.g.: 

Author(s) (Year). Data from: aTRAM - automated target restricted assembly method: a fast method for assembling loci across divergent taxa from next-generation sequencing data. Dryad Digital Repository. doi:10.5072/FK2dryad.8s4d6


Check out the information for depositors at http://datadryad.org/pages/faq
- How do I refer to my Dryad data in my article?
- How can I see how often my data package is viewed and downloaded?
- How can I add my data package to a Data section on my CV?
- If my data files are embargoed, when will they be released?
- and more!

Thanks again for taking this important step to preserve and share your research data.
We welcome your feedback! Contact us at help@datadryad.org.

The Dryad Team
datadryad.org 
```

Does this work for you, @elizabethdryad ?

**Comment by elizabethdryad** [comment]
Yes, this is great! I can take or leave adding the title to the reference citation. DOI is most important

File: dspace/modules/api/src/main/java/org/dspace/workflow/WorkflowEmailManager.java [hunk 1]
```diff
@@ -113,17 +115,24 @@ private static void notifyOfApproval(Context c, Item i, String emailFilename)
...
+            if(manuscriptIdentifier.length() == 0) {
+                manuscriptIdentifier = "none available";
+            }
+            email.addArgument(manuscriptIdentifier);
 
-            email.addArgument(manuScriptIdentifier);
+            // add a DOI URL as well:
+            String doi_url = "";
+            Matcher doimatcher = Pattern.compile("doi:(.+)").matcher(datapackageDoi);
+            if (doimatcher.find()) {
```
Comment (ryscher) [inline dspace/modules/api/src/main/java/org/dspace/workflow/WorkflowEmailManager.java:132]: should log an error if the DOI is not found, and put replacement text in the email argument

**Comment by ryscher** [comment]
Looks good. Tested on dev server.

**Pull request merged**